internal/core/src/exec/expression/Expr.h

```diff
@@ -307,16 +307,17 @@ class SegmentExpr : public Expr {
 
     int64_t
     GetNextBatchSize() {
-        auto current_chunk = is_index_mode_ && use_index_ ? current_index_chunk_
-                                                          : current_data_chunk_;
-        auto current_chunk_pos = is_index_mode_ && use_index_
+        auto is_sealed = segment_->type() == SegmentType::Sealed;
+        auto current_chunk = is_index_mode_ && use_index_ && is_sealed
```

Comment:
why we add is_sealed check here?
because geometry index is the only scalar index built for growing?

can we make is_index_mode_ && use_index_ && is_sealed one single variable to avoid calculation every time